src-built-in/components/userPreferences/src/components/general/Toolbar.jsx

```diff
@@ -1,26 +1,58 @@
 import React from 'react';
+import Checkbox from '../checkbox';
 
 export default class Toolbar extends React.Component {
 	constructor(props) {
 		super(props);
 		this.state = {
-			monitor: "all"
+			monitor: "0",
+			toolbarType: "Floating",
+			change: false,
 		};
 		this.setMonitor = this.setMonitor.bind(this);
+		this.setToolbarState = this.setToolbarState.bind(this);
 		this.restartApplication = this.restartApplication.bind(this);
 	}
 
 	/**
 	 * Sets the monitor by calling the setPreferences API. This will override the config for the component.
+	 * Possible values:
+	 * All monitors: "all"
+	 * Previous monitor when floating (defaults to primary): "0"
 	 * @param {event} e
 	 */
-	setMonitor(e) {
-		let monitor = e.target.value;
+	setMonitor(value) {
 		this.setState({
-			monitor: monitor
+			monitor: value
 		});
-		FSBL.Clients.ConfigClient.setPreference({ field: "finsemble.components.Toolbar.window.monitor", value: monitor });
 		// There's no need to override spawnOnAllMonitors. Setting monitor to anything other than undefined will automatically override that value. Setting monitor="all" is equivalent.
+		FSBL.Clients.ConfigClient.setPreference({ field: "finsemble.components.Toolbar.window.monitor", value: value });		
+	}
+
+	/**
+	 * Sets the toolbar as Fixed or Floating by calling the setPreferences API. This will override the config for the component.
+	 * Triggers monitor position based on toolbar option
+	 * @param {event}
+	 */
+	setToolbarState() {
+		//Use the previous state to toggle the checkbox, Floating is always checked
+		let previousState = this.state.toolbarType;
+		let toolbarType;
+		
+		if (previousState === "Fixed") {
+			FSBL.Clients.ConfigClient.setPreference({ field: "finsemble.components.Toolbar.window.dockable", value: ["top", "bottom"] });
+			toolbarType = "Floating";
+			this.setMonitor("0");
+		} else if (previousState === "Floating") {
+			FSBL.Clients.ConfigClient.setPreference({ field: "finsemble.components.Toolbar.window.dockable", value: null });
+			toolbarType = "Fixed";
+			this.setMonitor("all");
+		}
+		
+		this.setState({
+			toolbarType: toolbarType,
```

Comment:
ComponentDidMount sets the default state that previousState is based off of, so it really shouldn't be empty. But I guess if junk gets put in the config file it could end up not getting set. I could default it there, or do something like toolbarType: toolbarType || "Floating" here.

Reply:
Actually this.state.ToolbarType defaults to "Floating" in the constructor, so there's no way it will be empty here. I like the other suggestion though.

Reply:
Ah right. Constructor is setting the state and `previousState` is pulling the current set state. So it in theory should always be defined as something usable.

That however raises one question here. Why is this method not named `toggleToolbarState` then?